### PR TITLE
Add `edge:add` and `edge:del` as trigger conditions, resolves SYN-5974

### PR DESCRIPTION
**Pull request by rakuy0**

### CODE REVIEW

File: docs/synapse/userguides/storm_ref_automation.rstorm [hunk 1]
```diff
@@ -359,6 +359,12 @@ is a dictionary which contains the following keys:
         ``$auto.opts.valu``
             The value of the triggering node.
 
+        ``$auto.opts.edge``
```
Comment (invisig0th) [inline docs/synapse/userguides/storm_ref_automation.rstorm:362]: We generally refer to this as "verb". Lets have a quick goback w/analysts about it :+1:

File: synapse/lib/storm.py [hunk 2]
```diff
@@ -65,6 +71,13 @@
 
     # Adds a tag #todo to every inet:ipv4 as it is tagged #aka
     trigger.add tag:add --form inet:ipv4 --tag aka --query {[ +#todo ]}
+
+    # Adds a tag #todo to the N1 node of every refs edge add
+    trigger.add edge:add --edge refs --query {[ +#todo ]}
```
Comment (invisig0th) [inline synapse/lib/storm.py:76]: may want to tweak this as well

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -999,6 +1012,8 @@
             ('--form', {'help': 'Form to fire on.'}),
             ('--tag', {'help': 'Tag to fire on.'}),
             ('--prop', {'help': 'Property to fire on.'}),
+            ('--edge', {'help': 'Edge name to fire on.'}),
```
Comment (invisig0th) [inline synapse/lib/storm.py:1015]: here too

File: synapse/lib/storm.py [hunk 4]
```diff
@@ -999,6 +1012,8 @@
...
+            ('--destform', {'help': 'Destination form to fire on.'}),
```
Comment (invisig0th) [inline synapse/lib/storm.py:1016]: perhaps `--n2form` here?

File: synapse/lib/trigger.py [hunk 5]
```diff
@@ -69,6 +71,14 @@
             'if': {'properties': {'cond': {'const': 'prop:set'}}},
             'then': {'required': ['prop']},
         },
+        {
+            'if': {'properties': {'cond': {'const': 'edge:add'}}},
+            'then': {'required': ['edge']},
+        },
+        {
+            'if': {'properties': {'cond': {'const': 'edge:del'}}},
+            'then': {'required': ['edge']},
```
Comment (invisig0th) [inline synapse/lib/trigger.py:80]: are we missing destform/n2form in this schema?

File: synapse/lib/trigger.py [hunk 6]
```diff
@@ -182,6 +198,82 @@ async def runTagDel(self, node, tag, view=None):
                 for _, trig in globs.get(tag):
                     await trig.execute(node, vars=vars, view=view)
 
+    async def runEdgeAdd(self, n1, edge, n2, view=None):
+        n1form = n1.form.name
+        n2form = n2.form.name
+        vars = {'auto': {'opts': {'edge': edge, 'n2': n2}}}
+        with self._recursion_check():
+            # try all the variations of the optional params
```
Comment (invisig0th) [inline synapse/lib/trigger.py:206]: lots of work being done here. any ideas on how it effects perf? perhaps an overall strategy for caching all the triggers for a given n1+verb+n2 would be warranted?

File: synapse/lib/trigger.py [hunk 7]
```diff
@@ -182,6 +198,82 @@ async def runTagDel(self, node, tag, view=None):
...
+            globs = self.edgeaddglobs.get((n1form, n2form))
+            if globs:
+                for _, trig in globs.get(edge):
+                    await trig.execute(n1, vars=vars, view=view)
+
+    async def runEdgeDel(self, n1, edge, n2, view=None):
+        n1form = n1.form.name
+        n2form = n2.form.name
+        vars = {'auto': {'opts': {'edge': edge, 'n2': n2}}}
+        with self._recursion_check():
```
Comment (invisig0th) [inline synapse/lib/trigger.py:244]: ( same about perhaps caching? )

File: docs/synapse/userguides/storm_ref_automation.rstorm (hunk 1 above)
Comment (rakuy0) [reply docs/synapse/userguides/storm_ref_automation.rstorm:362]: I went back and forth on this. `edge` felt more explicit that it now supports light edges, but verb is the right terminology. A goback sounds good :+1:

**Review by invisig0th — APPROVED** [review]

**Pull request merged**